example/example.cpp

```diff
@@ -0,0 +1,61 @@
+#include "osrm/json_container.hpp"
+#include "osrm/engine_config.hpp"
+#include "osrm/route_parameters.hpp"
+#include "osrm/osrm.hpp"
+
+#include <string>
+#include <utility>
+#include <iostream>
+#include <exception>
+
+int main(int argc, const char *argv[]) try
+{
+    if (argc < 2)
+    {
+        std::cerr << "Error: Not enough arguments." << std::endl
+                  << "Run " << argv[0] << " data.osrm" << std::endl;
+    }
```

Comment:
Eeeeek, early exit missing: `return EXIT_FAILURE;`, otherwise `argv[1]` below is undefined behavior!

Also no need for `endl`, `'\n'` is perfectly fine and `stderr` has auto-flush.